Adding configuration files

rogeruiz opened the pull request:
I've updated the JavaScript and CSS / Sass parts of the `DEVELOPERS` documentation @micahsaul added. Comments and discussion on the files I'm adding are welcome! Some of these are borrowed from `18F Guides` and others from the `ffd-microsite` project which should help with #2.

rogeruiz committed:
Add Scss and JS linting documentation

rogeruiz committed:
Add git ignorance file

rogeruiz committed:
Add EditorConfig settings and description

rogeruiz committed:
Reword `eslint` section to match `editor`

rogeruiz commented:
I didn't add a `gitignore` section to the `DEVELOPERS` documentation, but it seems like I should. :thought_balloon:

rogeruiz commented on .editorconfig:
```diff
@@ -0,0 +1,17 @@
+# IDE settings for the Midas project. Many IDE's support this as a starting point
```
I need to clean up the comments in here. _Yikes_!

rogeruiz commented on DEVELOPERS.md:
```diff
@@ -32,14 +32,38 @@ Code review is an important part of development, and should not be viewed as a c
+character is a `space` or a `tab`. [EditorConfig] [editor-config] is a python
+library which helps standardize these settings between various editors in one
+place. This repository includes a generic `.editorconfig` file. Use this file to
+setup your text editors across your team.
```
Maybe mentioned that this should be a starting point across the team rather than the baseline for the team.

rogeruiz commented on DEVELOPERS.md:
```diff
@@ -32,14 +32,38 @@ Code review is an important part of development, and should not be viewed as a c
+#### JavaScript
+
+For working with JavaScript on the server, use a version of NodeJS greater than
+or equal to `4.0.0`. When using `npm`, use a version greater than or equal to
```
This version of `node` is super out of date already. This should be updated to `5.5.0`.

juliasolorzano commented on .gitignore:
```diff
@@ -0,0 +1,41 @@
+# OS and Editor files
+#
+.DS_Store
+*.DS_Store
```
Is there a reason why we need both `.DS_Store` and `*.DS_Store`?

rogeruiz replied:
I don't actually remember. I've been using the same `.gitignore_global` to make these that I have just kept appending to it over the years.

juliasolorzano replied:
Truth. Totally cool with leaving in, but maybe just pick one.

juliasolorzano commented on DEVELOPERS.md:
```diff
@@ -32,14 +32,38 @@ Code review is an important part of development, and should not be viewed as a c
+Use `eslint` to lint your JavaScript files and enforce code style and
+conventions for the team. This repository includes a generic `.eslintrc` file.
+Use this file to setup your JavaScript linting across your team.
+
```
What do you think about also pointing them to the AirBNB JS style guide? It's something the Frontend Guild has agreed to and would be good for us to start practicing: https://github.com/airbnb/javascript

rogeruiz replied:
I'll update the documentation around how to get this to work since the airbnb style guide is an `npm` package that the project would need to install in order to use it. Do you think we should keep in a generic one for `ffd` projects that don't use node? Not thinking of any off the top of my head, but the AirBNB guide needs `npm` in order to use it on a project.

juliasolorzano replied:
Good point! Let's move forward with using `npm` until we decide it's not useful to do so.

juliasolorzano commented:
This LGTM! Added a few notes and please let me know if you have any questions.

rogeruiz committed:
Remove previous project

rogeruiz committed:
Remove unnecessary line

rogeruiz committed:
Update to a non-ancient version of Node

juliasolorzano commented on .gitignore:
```diff
@@ -0,0 +1,41 @@
+Thumbs.db
+*.sublime-project
+*.sublime-workspace
+.svn
```
Not sure this is necessary?

rogeruiz committed:
Remove other VCS

rogeruiz committed:
Update eslint recommendations;

This patch updates the recommendations for eslint to reflect the latest
push from the Front End Guild to leverage the Airbnb JavaScript style
guide. 18F/frontend#114

rogeruiz committed:
Update language to be more of a suggestion

rogeruiz commented:
🎉 @juliaelman This PR is now updated with all the useful things we touched upon in the comments above ☝️

juliasolorzano commented:
Perfect! :rocket:
